docs/cryptography/contributionSigning.md

```diff
@@ -0,0 +1,126 @@
+# Contribution Signing
+
+In order to verify that contributions listed in the `transcript.json` are actually generated by the addresses listed in the transcript, the contributors MAY sign their contribution with their Ethereum address that they logged in with. And in turn, the secret contributed to the ceremony can be used as a BLS private key which can then sign the participant's Ethereum address or their GitHub name.
+
+## Contribution
+
+### EIP 712
+
+[EIP 712](https://eips.ethereum.org/EIPS/eip-712) defines a standard for messages signed with Ethereum ECDSA keys. This EIP is made available by most JSON RPCs/Web3 Providers under the `eth_signTypedData`/`web3.eth.signTypedData` endpoint. It requires `types`, `primaryType`, `domain`, and `message` combined into a single JSON object. Only the `message` should need to be changed for each contribution, the rest are static fields required by EIP712. These objects are defined as follows:
+
+### `types`
+
+```JSON
+"types": {
+    "EIP712Domain": [
+        {"name":"name","type":"string"},
+        {"name":"version","type":"string"},
+        {"name":"chainId","type":"uint256"}
+    ],
+    "contributionPubkey": [
+        {"name": "numG1Powers", "type": "uint256"},
+        {"name": "numG2Powers", "type": "uint256"},
+        {"name": "potPubkey", "type": "bytes"}
+    ],
+    "PoTPubkeys": [
+        { "name": "potPubkeys", "type": "contributionPubkey[]"}
+    ]
+}
+```
+
+### `primaryType`
+```JSON
+"primaryType": "PoTPubkeys",
+```
+
+### `domain`
+
+```JSON
+"domain": {
+    "name": "Ethereum KZG Ceremony",
+    "version": "1.0",
+    "chainId": 1
+}
+```
+
+Note that the array of `contributionPubkey[]` MUST be sorted in increasing order of `numG1Powers` with ties broken by increasing `numG2Powers`. 
+
+### `message`
+
+`message` contains the string-hex (including leading `0x`) encoded PoT-Pubkeys taken from [contribution.md](../contribution/contribution.md). These are `bls.G2Point`s corresponding to the setups for each of the numbers of powers. Thus, an example `message` corresponding with {`1`, `1`, `1`, `1`} (note this is just for demonstration purposes, these specs REQUIRE the secrets to be different) as secrets would be:
```

Comment:
"these specs REQUIRE the secrets to be different"

Just to add some notes: The sequencer checks this when they receive a contribution. Note, that they do not check whether the secret is distinct amongst all contributions (which is fine)

This means that a user can contribute once with secret (a,b,c,d) since (a,b,c,d) are all distinct, it will be accepted.

The user can then contribute again with (a,b,c,d) and it will be accepted.

Reply:
Yes, this is permissible behavior as it is a specific case of the more general making-your-keys-algebraically-related subcategory which we cannot detect. This is a part of our honesty assumption.

 The simple check we do do is to prevent faulty (but honest) contributions.